airbyte-lib/airbyte_lib/source.py

```diff
@@ -369,10 +369,13 @@ def read(self, cache: SQLCacheBase | None = None) -> ReadResult:
         if cache is None:
             cache = get_default_cache()
 
-        cache.register_source(source_name=self.name, source_catalog=self.configured_catalog)
+        cache.register_source(
+            source_name=self.name, incoming_source_catalog=self.configured_catalog
+        )
```

Comment:
Because sources can have a hundred or more streams, wdyt of passing the list of _selected_ streams (or all/None if nothing is selected). Since this registration occurs during "read()", we should know here which of the streams are actually going to be used and which will be ignored.

If we can implement without too much added complexity, I'd rather not store metadata on streams that are not going to receive data.

Reply:
This is already what happens - `configured_catalog` is filtered by the selected streams of the source: https://github.com/airbytehq/airbyte/blob/1268c3740b565aa773cf1643fbc98abc4d63c4a4/airbyte-lib/airbyte_lib/source.py#L177

Reply:
Nice! Thanks for confirming.